server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -39,15 +40,22 @@
         private final IndexSearcher searcher;
         private final LeafReaderContext readerContext;
         private final int docId;
-        private final SourceLookup sourceLookup = new SourceLookup();
```

Comment:
Interesting, so we were using a different SourceLookup instance for each hit? what does that mean in practice?

Reply:
I think reusing the `SourceLookup` here doesn't hurt or help, really. But it brings it closer to how the search phase works which is nice. We end up repositioning the `SourceLookup` before each fetch. And preloading the source too.

Reply:
indeed, so I believe that as long as all the sub fetch phases share the same source lookup for the same hit, we make sure that _source is loaded only once for that hit (as part of fetch, query is a different story)